src/validation.h

```diff
@@ -425,6 +425,12 @@ class CVerifyDB {
 /** Replay blocks that aren't fully applied to the database. */
 bool ReplayBlocks(const CChainParams& params, CCoinsView* view);
 
+inline CBlockIndex* LookupBlockIndex(const uint256& hash)
+{
+    BlockMap::const_iterator it = mapBlockIndex.find(hash);
```

Comment:
Missing assert lock is held.

Reply:
Missing `EXCLUSIVE_LOCKS_REQUIRED(cs_main)` too :-)